### PR TITLE
doc: migration guide

### PR DESCRIPTION
**Pull request by DenBeke**

### CODE REVIEW

File: website/docs/migration_guide.html.markdown [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+### Compute
+
+All the old compute resources have been regrouped under a new name: `Instance`. `Compute` is now the general branch name of all server related resources. In the future, baremetal and kapsule will become available in this branch.  
+This means that all old resources are now prefixed with `scaleway_compute_instance_`.
+
+#### `scaleway_server` -> `scaleway_compute_instance_server`
+
+`scaleway_server` was renamed to `scaleway_compute_instance_server`.
+
+In version 1, attachments of volumes where done on the volume resource. But from now on, this is done on the `scaleway_compute_instance_server` resources.
```
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: resource~~s~~.

File: website/docs/migration_guide.html.markdown [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+So to create a server with a volume attached:
```
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: "Therefore"?

File: website/docs/migration_guide.html.markdown [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+  root_volume {
+    delete_on_termination = false
+  }
+
+  additional_volume_ids = [ "${scaleway_compute_instance_volume.data.id}" ]
+}
+```
+
+#### `scaleway_compute_instance_ip` -> `scaleway_compute_instance_ip`
```
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: `scaleway_ip` -> ...

File: website/docs/migration_guide.html.markdown [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+`scaleway_ip` was renamed to `scaleway_compute_instance_ip` and the argument `server` was renamed to `server_id`.
+
+```hcl
+resource "scaleway_compute_instance_ip" "test_ip" {
+  server_id = "xxxx-xxxx-xxxx"
```
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`

File: website/docs/migration_guide.html.markdown [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+`scaleway_user_data` is now part of the `scaleway_compute_instance_server` resource.
+
+
+
+#### Removed: `scaleway_token`
+
+The `scaleway_token` was removed in version 2.
+
+Tokens should be created in the console.
```
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: Prefix: "For security reasons"?

File: website/docs/migration_guide.html.markdown [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+
+#### Removed: `scaleway_ssh_key`
+
+The `scaleway_ssh_key` was removed in version 2.
+
+SSH keys should be uploaded in the console.
```
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: Prefix: "For security reasons"?

File: website/docs/migration_guide.html.markdown (hunk 3 above)
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: "Renamed: ..."

File: website/docs/migration_guide.html.markdown [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+## Resources
+
+All resources are from now on prefixed by `scaleway`, their branch and their product name (`scaleway_{branch-name}_{product-name}_{resource-name}`). For instances an S3 bucket belongs to the `Storage` branch and is a resource of the `Object` product. Hence it is named: `scaleway_storage_object_bucket`.
+
+### Compute
+
+All the old compute resources have been regrouped under a new name: `Instance`. `Compute` is now the general branch name of all server related resources. In the future, baremetal and kapsule will become available in this branch.  
+This means that all old resources are now prefixed with `scaleway_compute_instance_`.
+
+#### `scaleway_server` -> `scaleway_compute_instance_server`
```
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: "Renamed: ..."

File: website/docs/migration_guide.html.markdown [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+`scaleway_ip` was renamed to `scaleway_compute_instance_ip` and the argument `server` was renamed to `server_id`.
+
+```hcl
+resource "scaleway_compute_instance_ip" "test_ip" {
+  server_id = "xxxx-xxxx-xxxx"
+}
+```
+
+
+#### `scaleway_volume` -> `scaleway_compute_instance_volume`
```
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: "Renamed: ..."

File: website/docs/migration_guide.html.markdown [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+The `scaleway_volume_attachment` was removed in version 2.
+
+Volumes can in version 2 only be attached on the server resource. The [above example](#scaleway_server-gt-scaleway_compute_instance_server) shows how this works.
+
+
+
+### Storage
+
+#### `scaleway_bucket` -> `scaleway_storage_object_bucket`
```
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: "Renamed: ..."

File: website/docs/migration_guide.html.markdown [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+The `scaleway_bucket` was moved to the `object` product in the `storage` branch.
+
+It's behaviour remained the same, but we also added an `acl` argument. This argument takes canned ACLs.
```
Comment (QuentinBrosse) [inline website/docs/migration_guide.html.markdown]: Link on `acl`

File: website/docs/migration_guide.html.markdown [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+layout: "scaleway"
+page_title: "Migration Guide"
+description: |-
+  Migrating your Scaleway provider from v1 to v2.
+---
+
+# Migrating from v1 to v2
+
+This page guides you through the process of migrating your version 1 resources to their version 2 equivalent.
+In version 2 we completely changed the naming of all resources, as well as their attributes.
```
Comment (jerome-quere) [inline website/docs/migration_guide.html.markdown]: Introduce some context like:
  - To prepare the launch of all new scaleway product we completely changed the naming....

File: website/docs/migration_guide.html.markdown [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+
+## Configuration
+
+To accomodate to the changes in the API and to be future proof, we decided to change the configuration management in version 2.
```
Comment (jerome-quere) [inline website/docs/migration_guide.html.markdown]: Something like:

In order to unify configuration management across all scaleway developer tools, we decided

File: website/docs/migration_guide.html.markdown [hunk 13]
```diff
@@ -8,11 +8,11 @@ description: |-
 # Migrating from v1 to v2
 
 This page guides you through the process of migrating your version 1 resources to their version 2 equivalent.
-In version 2 we completely changed the naming of all resources, as well as their attributes.
+To prepare the launch of all new Scaleway products, we completely changed the naming of all resources (as well as their attributes) in version 2 of the Terraform provider.
 
 ## Configuration
 
-To accomodate to the changes in the API and to be future proof, we decided to change the configuration management in version 2.
+In order to unify configuration management across all scaleway developer tools, we decided to change the configuration management in version 2.
```
Comment (jerome-quere) [inline website/docs/migration_guide.html.markdown]: nit:  we ~~decided to~~ change

**Pull request merged**